nemo/nemo/core/neural_modules.py

```diff
@@ -363,24 +376,6 @@ def local_parameters(self) -> Optional[Dict]:
         """
         return self._local_parameters
 
-    @property
-    def input_ports(self) -> Optional[Dict[str, NeuralType]]:
```

Comment:
keep these instead of your newly introduced input_port_definitions and output_port_definitions

Reply:
Please find my comment above.